.github/workflows/reusable-solidity-docs.yml

```diff
@@ -0,0 +1,248 @@
+      - name: Build Markdown docs
+        run: yarn run hardhat docgen
+
+      - name: Add Table of Contents
+        if: inputs.addTOC == true
+        run: |
+          yarn global add markdown-toc 
+          sed -i '2s/^/\<!-- toc --\> \n/' ./generated-docs/index.md
+          sed -i '2s/^/## Table Of Contents \n/' ./generated-docs/index.md
+          markdown-toc ${{ inputs.tocOptions }} ./generated-docs/index.md
+
+      - name: Export artifacts
+        if: inputs.exportAsGHArtifacts == true
+        uses: actions/upload-artifact@master
+        with:
+          name: contracts-final-output
+          path: .${{ inputs.projectSubfolder }}/generated-docs
+
+      - name: Post link to artifacts in PR comment
+        if: |
+          inputs.exportAsGHArtifacts == true
+            && inputs.commentPR == true
+            && startsWith(github.ref, 'refs/pull')
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Solidity API documentation preview available in the artifacts of the https://github.com/${{ github.repository}}/actions/runs/${{ github.run_id}} check.'
+            })
+
+      - name: Import GPG key
+        if: inputs.publish == true && inputs.verifyCommits == true
+        uses: crazy-max/ghaction-import-gpg@111c56156bcc6918c056dbef52164cfa583dc549 # v5.2.0
+        with:
+          gpg_private_key: ${{ secrets.gpgPrivateKey }}
+          passphrase: ${{ secrets.gpgPassphrase }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
+      - name: Sync generated docs with the specified repository
+        if: inputs.publish == true
+        env:
+          API_TOKEN_GITHUB: ${{ secrets.githubToken }}
+        run: |
+          git config --global user.email ${{ inputs.userEmail }}
+          git config --global user.name ${{ inputs.userName }}
+          git clone --single-branch --branch ${{ inputs.destinationBranch }} \
+            https://x-access-token:$API_TOKEN_GITHUB@github.com/${{ inputs.destinationRepo }}.git \
+            dest-repo-clone
+          cd dest-repo-clone
+          mkdir -p ${{ inputs.destinationFolder }}
+          rsync -avh ${{ inputs.rsyncDelete && '--delete' || '' }} \
+            ../generated-docs \
+            ${{ inputs.destinationFolder }}
+          git add -A
+          if git status | grep -q "Changes to be committed"
+          then
+            git commit ${{ inputs.verifyCommits && '-S' || '' }} -m "Update docs by https://github.com/${{ github.repository}}/actions/runs/${{ github.run_id}}"
+            echo "Pushing git commit"
+            git push --set-upstream origin HEAD:${{ inputs.destinationBranch }}
+          else
+            echo "No changes detected"
+          fi
```

Comment:
Have you considered publishing the docs to a GCP bucket instead of pushing it directly in a commit to the destination repo?

A workflow in the destination repository could pull the docs it needs from the bucket.

Reply:
I did consider pushing to the bucket back in a day when I thought we'll host the docs in docs.keep.network. Back then I made the job generate the .html files and push them to the bucket. But then we decided to host everything in docs.threshold.network, and pushing it straight to the GitBook-synced repo seemed like a good and pretty simple idea (I did not realize back then than making the commits verified wouldn't be a straightforward thing).
Now that I have the verification figured out I don't see any obvious advantages of using the bucket approach. And one disadvantage of using the bucket approach is that it would be hard to setup the workflows in a way that the docs would be updated immediately once a new release is published (unless we'd configure the workflow in the GitBook repo to run on schedule very often).